app/src/frontend/data-extracts.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { FunctionComponent } from 'react';
+
+import { dateReviver } from '../helpers';
+
+interface ExtractViewModel {
+    extract_id: number;
+    extracted_on: Date;
+    download_path: string;
+}
+
+interface DataExtractsState {
+    latestExtract: ExtractViewModel;
+    previousExtracts: ExtractViewModel[];
+}
+
+export default class DataExtracts extends React.Component<{}, DataExtractsState> {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            latestExtract: undefined,
+            previousExtracts: undefined
+        };
+    }
+
+    async componentDidMount() {
+        const res = await fetch('/api/extracts');
+        const data = JSON.parse(await res.text(), dateReviver);
```

Comment:
Using `JSON.parse` with `dateReviver` to create `Date` objects, which `await res.json()` wouldn't do?

Reply:
(I could have used a one-line comment just to point this out as it's less familiar)

Reply:
I don't quite understand what you mean here, could you rephrase please?

Reply:
ah I think I understand now. Yes, this is to actually parse the JSON date strings into Date objects. Sure, I will aim to put comments in such places where it's slightly less obvious what's happening. Do you want them in the code or rather in a comment to the PR?

Reply:
Great :) yes, I meant that I was briefly puzzled by why the `dateReviver` function was necessary. A short explanatory comment in the code would help.